src/utils/media-utils.ts

```diff
@@ -0,0 +1,80 @@
+import axios from "axios"
+
+import { config } from "@config/config"
+
+import { getAccessToken } from "@root/utils/token-retrieval-utils"
+
+const GITHUB_ORG_NAME = config.get("github.orgName")
+
+type ItemType = "dir" | "file"
+
+export interface MediaFile {
+  name: string
+  type: ItemType
+  sha: string
+  path: string
+}
+
+interface MediaFileInput {
+  file: MediaFile
+  siteName: string
+  directoryName: string
+  mediaType: string
+  isPrivate: boolean
+}
+
+interface MediaFileOutput {
+  name: string
+  sha: string
+  mediaUrl: string
+  mediaPath: string
+  type: ItemType
+}
+
+interface MediaDirOutput {
+  name: string
+  type: ItemType
+}
+
+export const isMediaFileOutput = (
+  t: MediaFileOutput | MediaDirOutput
+): t is MediaFileOutput => (t as MediaFileOutput).sha !== undefined
+
+export const getMediaFileInfo = async ({
+  file,
+  siteName,
+  directoryName,
+  mediaType,
+  isPrivate,
+}: MediaFileInput): Promise<MediaFileOutput | MediaDirOutput> => {
+  if (file.type === "dir") {
+    return {
+      name: file.name,
+      type: "dir",
+    }
+  }
```

Comment:
this method feels too bloated alr - we are trying to smush file/dir retrieval into one method, which doesn't lead itself well to future extension.

as the caller already knows the file type here, i don't see why we can't just separate this into `getMediaFileInfo` and `getMediaDirectoryInfo`. this also makes it more apparent that the media directory info relies only on the type and the name.

Reply:
Yeah that makes sense, I've removed the directory info portion and moved that directly into the retrieval method, since the check alone basically already gives the return output - https://github.com/isomerpages/isomercms-backend/pull/740/commits/2d69f03b73cb67e275eeceaf394674ea9031def9